attributes-info/src/compiler_features.rs

```diff
@@ -0,0 +1,91 @@
+use std::collections::HashMap;
+use std::fs::File;
+use std::io;
+use std::io::Write;
+
+use serde::{Deserialize, Serialize, Serializer};
+use serde::ser::SerializeSeq;
+use serde_json::ser::{Formatter, PrettyFormatter};
+use version_check::Version;
+
+use rustc_feature::{ACCEPTED_FEATURES, ACTIVE_FEATURES, Feature, Features, REMOVED_FEATURES, STABLE_REMOVED_FEATURES, State};
+
+use crate::formatter::MyFormatter;
+
+#[derive(Serialize, Deserialize)]
+struct FeatureInfo {
+    name: String,
+    state: String,
+    since: String
+}
+
+impl FeatureInfo {
+    fn from(f: &Feature) -> Self {
+        FeatureInfo {
+            name: f.name.as_str().to_string(),
+            state: state_str(f),
+            since: f.since.to_string()
+        }
+    }
+}
+
+pub(crate) fn update_compiler_features(compiler_features_json_path: &str) {
```

Comment:
Maybe
```suggestion
pub(crate) fn update_compiler_features_json(compiler_features_json_path: &str) {
```
 to be consistent with `builtin_attributes::generate_builtin_attributes_json`